src/main/java/sg/dex/starfish/impl/memory/MemoryJob.java

```diff
@@ -70,14 +74,23 @@ public boolean isCancelled() {
 	@Override
 	public Map<String,Object> get(long timeoutMillis, TimeUnit timeUnit) {
 		try {
+			this.setStatus(Status.succeeded);
 			return future.get(timeoutMillis, timeUnit);
 		}
 		catch (InterruptedException | ExecutionException | TimeoutException e) {
 			Throwable cause=e.getCause();
+			this.setStatus(Status.failed);
 			throw new JobFailedException("Job failed with exception: "+cause,e);
 		}
 	}
 
+	@Override
+	public Job.Status getStatus() {
+		return this.status;
+	}
+	public void setStatus( Job.Status status) {
+		this.status = status;
```

Comment:
Needs protection. You shouldn't be able to change the status of a "failed" or "succeeded" job, for example.